src/App.tsx

```diff
@@ -16,35 +16,104 @@ export const goodsFromServer = [
 ];
 
 export const App: React.FC = () => {
+  const [goods, setGoods] = useState<string[]>([...goodsFromServer]);
+  const [isAlphabetical, setIsAlphabetical] = useState<boolean>(false);
+  const [isByLength, setIsByLength] = useState<boolean>(false);
+  const [isReversed, setIsReversed] = useState<boolean>(false);
+
+  const reverseButtonRef = useRef<HTMLButtonElement | null>(null);
+
+  const sortAlphabetically = (): void => {
```

Comment:
This function can be moved outside of component, just provide arguments for passing in array, sort value and isReverse.